specs/labels-and-localization/README.md

```diff
@@ -0,0 +1,188 @@
+- Client-provided icon elements in the table (e.g. icons as the primary content of table headers): Follows the same guidance in [accessibility.md](../../packages/nimble-components/docs/accessibility.md), i.e. the icons should provide accessible text via the `title` attribute (and clients will handle localizing it themselves).
+- (Mapping/ Icon Columns: As currently specced, `nimble-table-column-icon`'s `nimble-mapping-icon` has a `label` attribute which will become the icon `title`)
+
+## Localization
+
+Our general approach to localization is that clients will handle localizing strings. That has led us to add attributes on some Nimble components for clients to provide localized labels, but that doesn't scale well to components with many labels like the table.
+
+### Plan for Nimble-Provided Labels
+
+Conceptually the localized strings are very similar to the sharing pattern of Design Tokens. The vast majority of the time, you want to use the same value (i.e. numeric increment/decrement button text or collapse button text is not control instance specific) but want to adjust to a global config, i.e. theme or language. But also want to be able to override for specific controls as needed.
+
+We will create non-CSS-property design tokens, similar to the `theme` and `direction` design tokens. The values can be set via the `nimble-theme-provider`:
+
+```html
+<nimble-theme-provider
+    label-number-field-decrement="Increment"
+    label-number-field-increment="Decrement"
+    label-table-collapse-all="Collapse All"
+>
+<nimble-table></nimble-table>
+</nimble-theme-provider>
+```
+
+Pros:
+
+- Easy discovery of localized string configuration. They are all on the theme-provider.
+- Can configure once for the whole page in one spot, does not need to be per control. Much fewer duplicated attributes / slotted elements in the page.
+- Can override for specific elements as needed by wrapping in a theme-provider.
+- Can configure to be attribute or slotted on the theme provider. But if don't need mixed content then attribute makes sense.
+- User provided control-specific content should still be provided via attributes / slots. This is for control provided content (the table control defines the collapse button, not the user).
+
+Cons: 
+
+- Still somewhat verbose in the page, to have all of the localized strings as attributes
+    - We can consider using the  `fromView` attribute mode on the attributes for the strings on the theme-provider. If clients set the strings using the theme-provider properties rather than as attributes in the HTML (as is currently proposed for Angular), `fromView` would not reflect those strings back to the DOM.
+- Not an originally intended use case for DesignToken
+- Only nimble-components defined within Nimble can add new attributes to the theme-provider.
```

Comment:
Theoretically the `DesignToken`s could be set directly since they'd be exported from nimble-components, the same way all of our current DesignTokens are.

I guess non-Nimble code could also create DesignTokens for their strings/labels if they wanted to. Then wherever in their app they use the theme-provider, they could also have code to set their own token values on the theme provider.

I think defining the attributes on theme-provider are mainly a convenience, because that's where we want them to be set. The point of going with this approach would be that the strings could just be set once (on a theme-provider or global-theme-provider). So without exposing the attributes/properties, client code would have to import references to the individual DesignTokens for the labels, and call `token.setValueFor(themeProvider)` for each. And they'd have to get the English string to be localized somehow.  
I think defining the attributes/properties on the theme provider doesn't add much implementation cost (the prototype does it already), and it gives us lots of flexibility on ways to set the label values from Angular/Blazor.

Reply:
Hmmm what if we had a separate localization provider element? But instead of doing the same thing as theme provider, it does `DesignToken.setValueFor()` on its first parent theme provider (or maybe more precisely finds the first parent theme-provider and asks it what the target is since theme-provider and global-theme-provider have different targets):

```html
<nimble-theme-provider>
    <nimble-i18n
         number-field-increment="Blah"
         number-field-decrement="Blah"
    ></nimble-i18n>
</nimble-theme-provider>
```

Then we could split them up into more component providers or even "logical groupings" shared across components, and external systems can plugin their own behavior (implement new i18n-providers by extending a base class or something). We don't need a monolithic element for all components.

```html
<nimble-theme-provider>
    <nimble-i18n-numeric
         increment="Blah"
         decrement="Blah"
    ></nimble-i18n-numeric>
    <nimble-i18n-banner
         dismiss="Blah"
    ></nimble-i18n-banner>
    <nimble-i18n-table
         ...
    ></nimble-i18n-table>
</nimble-theme-provider>
```

Pros:
- Avoid polluting the set of properties to configure on the theme provider. Addresses the name spacing concern too in the other thread
- Think it cleans up some of the verbosity of the attribute names since they become target specific
- Avoids a scenario that every app has to pay the cost of having all the defined localization strings pulled into their build no matter what component they import. Avoid having just a button as a dependency adding all the table localization strings to the build output.
- Plugins / custom components can follow the same pattern as nimble components. Think it actually ends up working pretty darn well. Don't even need to worry about conflicts in attribute names (get your own element to define attributes) or css custom property names (like the theme and direction config tokens, these should not emit as CSS custom properties).

Cons: 
- If we split it up per component does that potentially result in a bunch of stuff to keep track of for users? At most it's one per component right, might not be that bad? It's still the exact same number of properties just moving them to different attributes. Maybe not really a con but I'll leave it here

Reply:
Updated the HLD based on our discussions. Current proposal is to add `nimble-i18n-core` and `nimble-i18n-table` as the 2 providers. Moved the proposal with everything on the theme-provider to Alternatives.